pandas/tools/merge.py

```diff
@@ -894,13 +896,21 @@ def get_result(self):
             if self.axis == 0:
                 new_data = com._concat_compat([x.values for x in self.objs])
                 name = com._consensus_name_attr(self.objs)
-                return Series(new_data, index=self.new_axes[0], name=name).__finalize__(self, method='concat')
+                if self._is_sp_series:
```

Comment:
That was my first approach but I also wanted to keep the symmetry between `axis = 1` and `axis = 0`. And if you want to check fill values etc, the case work seems necessary (if ugly) without refactoring.